### PR TITLE
Update the FAQ and add doc on versioning

### PR DESCRIPTION
**Pull request by jackkoenig**
### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [NA] Did you add at least one test demonstrating the PR?
- [NA] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

 - documentation   

#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

 - Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [x] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.0, 3.5.0) ?
- [x] Did you review?
- [x] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by mwachs5** [comment]
Nice! I have only looked at the raw view but content is good. Haven't checked how the table looks rendered

File: docs/src/appendix/versioning.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (mwachs5) [inline docs/src/appendix/versioning.md:1]: is there a corresponding website PR to add this to the menu.yml?

**Comment by mwachs5** [comment]
realized in all my refactoring I didn't actually fix https://github.com/freechipsproject/www.chisel-lang.org/issues/102 ... now this does. 🤦‍♀️

File: docs/src/explanations/modules.md [hunk 2]
```diff
@@ -111,16 +111,16 @@ import chisel3.{RawModule, withClockAndReset}
 class Foo extends Module {
   val io = IO(new Bundle{
     val a = Input(Bool())
-    val b = Output(Bool())
+    val b = Output(Bool()
```
Comment (jackkoenig) [inline docs/src/explanations/modules.md:114]:
```suggestion
    val b = Output(Bool())
```

**Comment by jackkoenig** [comment]
> Nice! I have only looked at the raw view but content is good. Haven't checked how the table looks rendered

You can view it in the diff: https://github.com/chipsalliance/chisel3/pull/1827/files?short_path=6959c6b#diff-6959c6be40448c233181770b98e858bacebc4a359abc27d6d06526aaaa30f23b

**Comment by mwachs5** [comment]
I am happy to approve this but I don't actually know how to properly apply the labels and such

**Review by mwachs5 — APPROVED** [review]

File: docs/src/appendix/versioning.md (hunk 1 above)
Comment (jackkoenig) [reply docs/src/appendix/versioning.md:1]: I was just planning to do it when opening the PR that bumps to include this.

**Pull request merged**